File: scripts/scripts/2bws.py
# -*- coding: utf-8 -*-

import os

total = "xprop -root _NET_NUMBER_OF_DESKTOPS | sed -e 's/_NET_NUMBER_OF_DESKTOPS(CARDINAL) = //'"
current = "xprop -root _NET_CURRENT_DESKTOP | sed -e 's/_NET_CURRENT_DESKTOP(CARDINAL) = //'"

total = int(os.popen(total).read())
current = int(os.popen(current).read())

wstr = [ '-','-','-','-','-' ]

for i in range(total):
    if i == current:
        wstr[i] = '✕'
    else:
        wstr[i] = '-'

wstr = '  '.join(wstr)
print(wstr)

File: scripts/scripts/2bws.cpp
#include <iostream>
#include <fstream>

using namespace std;

int main(){
	int wsno;
	system("xprop -root _NET_CURRENT_DESKTOP | sed -e 's/_NET_CURRENT_DESKTOP(CARDINAL) = //' > ws.txt");

	fstream x;
	char ws;

	x.open("ws.txt", ios::in);
	x >> ws;

	ws -= 47;

	// assign tag names
	switch(ws){
		case 1:
			cout<<"www";
			break;
		case 2:
			cout<<"irc";
			break;
		case 3:
			cout<<"code";
			break;
		case 4:
			cout<<"music";
			break;
		case 5:
			cout<<"misc";
			break;
		default:
			cout<<"hidden";
	}

	x.close();

	return 0;
}
